File: src/bytecode/structs/opcodes/OpJmp.ts
import { Instruction } from "../Instruction";
import { Opcode } from "../Opcode";

export class OpJmp extends Opcode {
  instruction: Instruction;

  constructor(instruction: Instruction) {
    super();

    this.instruction = instruction;
  }

  getObfuscated(): string {
    throw new Error("Method not implemented.");
  }
}

File: src/bytecode/structs/opcodes/op_jmp.rs
use crate::bytecode::structs::{instruction::Instruction, opcode::Opcode};

pub struct OpJmp {
    instruction: Instruction,
}

impl OpJmp {
    pub fn new(instruction: Instruction) -> Self {
        Self { instruction }
    }
}

impl Opcode for OpJmp {
    fn get_instruction(&self) -> &Instruction {
        &self.instruction
    }

    fn get_obfuscated(&self) -> &str {
        "pc = pc + inst.sBx"
    }

    fn is_valid(&self) -> bool {
        todo!()
    }
}
